### ISSUE TITLE
[BUG] On centos7(xfs), vm can't boot after backup and restore

### ISSUE THREAD

**Issue by futuretea** [harvester/harvester#849]
**Describe the bug**

After restore， vnc console show
![image](https://user-images.githubusercontent.com/15064560/117763829-0f66d700-b25e-11eb-80b4-0f6a51daf505.png)
or
![image](https://user-images.githubusercontent.com/15064560/117763899-2b6a7880-b25e-11eb-92c0-76dc8ea51c25.png)

**To Reproduce**
1. centos 7(xfs)+docker+kubeadm cluster
1. install harvester
1. create a vm
1. backup a vm to nfs
1. restore vm

**Expected behavior**

after vm restored， vm can boot correctly

**Support bundle**

**Environment:**
 - Harvester ISO version:  NO ISO, app-mode
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**
```bash
# kubectl get pod -o custom-columns='NAME:metadata.name,IMAGES:spec.containers[*].image' -A
NAME                                                   IMAGES
virt-launcher-b-new-zlfz2                              registry.suse.com/harvester-beta/virt-launcher:0.40.0-2
virt-launcher-b-qjw26                                  registry.suse.com/harvester-beta/virt-launcher:0.40.0-2
cdi-apiserver-bfd77bd48-jnhr5                          registry.suse.com/harvester-beta/cdi-apiserver:1.30.0
cdi-deployment-c9fbd68b7-99c68                         registry.suse.com/harvester-beta/cdi-controller:1.30.0
cdi-operator-6dd7745d77-d9k5c                          registry.suse.com/harvester-beta/cdi-operator:1.30.0
cdi-uploadproxy-68b69f7cdf-2pqv8                       registry.suse.com/harvester-beta/cdi-uploadproxy:1.30.0
harvester-5d878c868d-5dpjl                             rancher/harvester:master-head
harvester-5d878c868d-9d78j                             rancher/harvester:master-head
harvester-5d878c868d-txpbm                             rancher/harvester:master-head
harvester-network-controller-manager-c6df5447b-7mwhl   rancher/harvester-network-controller:master-head
harvester-network-controller-manager-c6df5447b-kn4nv   rancher/harvester-network-controller:master-head
harvester-network-controller-wljpf                     rancher/harvester-network-controller:master-head
virt-api-789cdfb468-9ctj7                              registry.suse.com/harvester-beta/virt-api:0.40.0-2
virt-api-789cdfb468-m2l7l                              registry.suse.com/harvester-beta/virt-api:0.40.0-2
virt-controller-7db4947c66-pmwxv                       registry.suse.com/harvester-beta/virt-controller:0.40.0-2
virt-controller-7db4947c66-zvdmx                       registry.suse.com/harvester-beta/virt-controller:0.40.0-2
virt-handler-t2zcw                                     registry.suse.com/harvester-beta/virt-handler:0.40.0-2
virt-operator-6f7dfb68bb-l49ls                         registry.suse.com/harvester-beta/virt-operator:0.40.0-2
coredns-74ff55c5b-m9pnz                                k8s.gcr.io/coredns:1.7.0
coredns-74ff55c5b-v9c5j                                k8s.gcr.io/coredns:1.7.0
etcd-master                                            k8s.gcr.io/etcd:3.4.13-0
kube-apiserver-master                                  k8s.gcr.io/kube-apiserver:v1.20.6
kube-controller-manager-master                         k8s.gcr.io/kube-controller-manager:v1.20.6
kube-flannel-ds-d24st                                  quay.io/coreos/flannel:v0.14.0-rc1
kube-proxy-pt2ng                                       k8s.gcr.io/kube-proxy:v1.20.6
kube-scheduler-master                                  k8s.gcr.io/kube-scheduler:v1.20.6
metrics-server-6f64b77899-qg822                        k8s.gcr.io/metrics-server/metrics-server:v0.4.4
multus-ds-amd64-6xddc                                  nfvpe/multus:v3.6
snapshot-controller-9f68fdd9-b2fd4                     k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0
snapshot-controller-9f68fdd9-qrx47                     k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0
backing-image-manager-17ed-8122                        longhornio/backing-image-manager:v1_20210422
csi-attacher-7865c49fbd-2t4vw                          longhornio/csi-attacher:v2.2.1-lh1
csi-attacher-7865c49fbd-cb4kp                          longhornio/csi-attacher:v2.2.1-lh1
csi-attacher-7865c49fbd-fl5f6                          longhornio/csi-attacher:v2.2.1-lh1
csi-provisioner-fdd754d5d-85mvn                        longhornio/csi-provisioner:v1.6.0-lh1
csi-provisioner-fdd754d5d-nsf6f                        longhornio/csi-provisioner:v1.6.0-lh1
csi-provisioner-fdd754d5d-xz4xn                        longhornio/csi-provisioner:v1.6.0-lh1
csi-resizer-d9db74686-6pdcd                            longhornio/csi-resizer:v0.5.1-lh1
csi-resizer-d9db74686-brgtp                            longhornio/csi-resizer:v0.5.1-lh1
csi-resizer-d9db74686-pfjjn                            longhornio/csi-resizer:v0.5.1-lh1
csi-snapshotter-7fbc7bdff-fkrg5                        longhornio/csi-snapshotter:v2.1.1-lh1
csi-snapshotter-7fbc7bdff-ncwkq                        longhornio/csi-snapshotter:v2.1.1-lh1
csi-snapshotter-7fbc7bdff-nvwd5                        longhornio/csi-snapshotter:v2.1.1-lh1
engine-image-ei-611d1496-g44qs                         longhornio/longhorn-engine:v1.1.1
instance-manager-e-6aebd2e1                            longhornio/longhorn-instance-manager:v1_20201216
instance-manager-r-4e6bf064                            longhornio/longhorn-instance-manager:v1_20201216
longhorn-csi-plugin-v7zqs                              longhornio/csi-node-driver-registrar:v1.2.0-lh1,longhornio/longhorn-manager:v1.1.1
longhorn-driver-deployer-7d4b5b75cb-khvdg              longhornio/longhorn-manager:v1.1.1
longhorn-manager-4jvx4                                 longhornio/longhorn-manager:v1.1.1
longhorn-ui-b4fb6fc48-js2fj                            longhornio/longhorn-ui:v1.1.1
```

**Comment by futuretea** [comment]
link to https://github.com/longhorn/longhorn/issues/2503

**Comment by guangbochen** [comment]
**To Test:**
This should be addressed after bumping the longhorn to v1.1.2 with https://github.com/longhorn/longhorn/issues/2503, the filesystem type of the disk should be `xfs`. U may test using the multi-disk feature from longhorn by adding an additional `xfs` disk and run VM backup and restore using this external disk.

**Comment by juadk** [comment]
Tested successfully with the last master branch `master-6ae8d8ee-head` and longhorn v1.1.2. ISO mode with a 5 nodes cluster.
1. Define NFS as backup-target
2. Create a centos VM with root xfs disk and add one additional xfs disk.
3. Populate xfs disks with data
4. Check that my NFS remote folder contains the volumes (also in a xfs partition)
5. Take a backup
6. Remove some files created
7. Shut down the VM
8. Restore from the snapshot
9. The VM booted successfully and all the data was there in both disks
